storage/pebble/bootstrap_test.go

```diff
@@ -0,0 +1,259 @@
+package pebble
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/engine/execution/state"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/testutils"
+	"github.com/onflow/flow-go/ledger/complete/mtrie/trie"
+	"github.com/onflow/flow-go/ledger/complete/wal"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage/pebble/registers"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+func TestRegisterBootstrap_NewBootstrap(t *testing.T) {
+	t.Parallel()
+	sampleDir := path.Join(unittest.TempDir(t), "checkpoint.checkpoint")
+	rootHeight := uint64(1)
+	log := zerolog.New(io.Discard)
+	cache := pebble.NewCache(1 << 20)
+	defer cache.Unref()
+	opts := DefaultPebbleOptions(cache, registers.NewMVCCComparer())
+	p, dir := unittest.TempPebbleDBWithOpts(t, opts)
+	// set heights
+	require.NoError(t, p.Set(firstHeightKey(), encodedUint64(rootHeight), nil))
+	require.NoError(t, p.Set(latestHeightKey(), encodedUint64(rootHeight), nil))
+	// errors if FirstHeight or LastHeight are populated
+	_, err := NewRegisterBootstrap(p, sampleDir, rootHeight, log)
+	require.ErrorContains(t, err, "cannot bootstrap populated DB")
+	require.NoError(t, os.RemoveAll(dir))
+}
+
+func TestRegisterBootstrap_IndexCheckpointFile_Happy(t *testing.T) {
+	t.Parallel()
+	log := zerolog.New(io.Discard)
+	rootHeight := uint64(10000)
+	unittest.RunWithTempDir(t, func(dir string) {
+		tries, registerIDs := simpleTrieWithValidRegisterIDs(t)
+		fileName := "simple-checkpoint"
+		require.NoErrorf(t, wal.StoreCheckpointV6Concurrently(tries, dir, fileName, log), "fail to store checkpoint")
+		checkpointFile := path.Join(dir, fileName)
+
+		cache := pebble.NewCache(1 << 20)
+		opts := DefaultPebbleOptions(cache, registers.NewMVCCComparer())
+		defer cache.Unref()
+		pb, dbDir := unittest.TempPebbleDBWithOpts(t, opts)
+
+		bootstrap, err := NewRegisterBootstrap(pb, checkpointFile, rootHeight, log)
+		require.NoError(t, err)
+		err = bootstrap.IndexCheckpointFile(context.Background())
+		require.NoError(t, err)
+
+		// create registers instance and check values
+		reg, err := NewRegisters(pb)
+		require.NoError(t, err)
+
+		require.Equal(t, reg.LatestHeight(), rootHeight)
+		require.Equal(t, reg.FirstHeight(), rootHeight)
+
+		for _, register := range registerIDs {
+			val, err := reg.Get(*register, rootHeight)
+			require.NoError(t, err)
+			require.Equal(t, val, []byte{uint8(0)})
+		}
+
+		require.NoError(t, pb.Close())
+		require.NoError(t, os.RemoveAll(dbDir))
```

Comment:
nit: this is already done by `RunWithTempDir`

Reply:
this one is intentional, the `dir` file is used for storing the checkpoint file (that `RunWithTempDir ` will close) and the `dbDir` is for the pebble instance